src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -764,7 +824,7 @@ private static TypeSymbol GetNamedPropertyType(ObjectType baseType, IPositionabl
 
                     if (writeOnlyDiagnostic.Level == DiagnosticLevel.Error)
                     {
-                        return new ErrorTypeSymbol(Enumerable.Empty<ErrorDiagnostic>());
+                        return new UnassignableTypeSymbol(Enumerable.Empty<ErrorDiagnostic>());
```

Comment:
Here's a scenario where we return an error with no diags (since I remember you had a comment asking if there were any such scenarios)